custom_components/keymaster/helpers.py

```diff
@@ -147,7 +147,12 @@ async def async_update_zwave_js_nodes_and_devices(
     child_locks: List[KeymasterLock],
 ) -> None:
     """Update Z-Wave JS nodes and devices."""
-    client = hass.data[ZWAVE_JS_DOMAIN][entry_id][ZWAVE_JS_DATA_CLIENT]
+    try:
+        zwave_entries = hass.config_entries.async_entries(ZWAVE_JS_DOMAIN)
+        zwave_entry = zwave_entries[0] if zwave_entries else None
```

Comment:
I think we may need to loop and check the states to look for `ConfigEntryState.LOADED` and use the loaded entry.

Reply:
Maybe something like:

```python
zwave_entry = entry for entry in zwave_entries if entry.state == ConfigEntryState.LOADED
```

I'm not 100% sure this code is usable but you get the idea.

Reply:
```python
zwave_loaded_entries = [entry for entry in zwave_entries if entry.state == ConfigEntryState.LOADED]
zwave_entry = zwave_loaded_entries[0] if zwave_loaded_entries else None
```

This is what it needs to change to as the ```x for x in y``` syntax is a list comprehension (i tried it outside of a list and python did not like it :laughing: )

Also

```python
from homeassistant.config_entries import ConfigEntryState
```
needs to be added to the imports.